### ISSUE TITLE
I2C Example?

### ISSUE THREAD

**Issue by JasonKleban** [newAM/libftd2xx-rs#14]
Thanks for making this library!

Complete noob to rust and electronics and microcontrollers here.  I'm trying to use the FT232H so my laptop can talk to a LSM303DLHC (Accelerometer + Compass) and some PCA9685 (Servo Motor Driver).  I chose these parts BECAUSE [they have rust crates](https://github.com/rust-embedded/awesome-embedded-rust) - I don't even have a real project in mind, I just want to learn about how to interact with the real world directly from my computer (no microcontroller) using Rust (not python or c, etc).  This is part of my self-directed educational journey, a step forward maybe of when I wrote [this related stackoverflow question](https://stackoverflow.com/questions/64814865/rust-from-desktop-to-i2c-or-spi-or-uart-while-maintaining-crate-compatibility).

I like the sound of the chainability and addressability of I2C, but I've never done anything like it before.  I sort of understand how this could be used to talk to something with low-level GPIO high and low signals, but I believe this also will let us use the higher-level device crates with the supported protocols to give each addressable device high-level commands.  Assuming I can figure out how to wire and power these components, would it be easy for you to add/share an example of how I can `libftd2xx-rs` to establish the bus or whatever for crates such as [pwm-pca9685](https://crates.io/crates/pwm-pca9685) and [lsm303dlhc](https://crates.io/crates/lsm303dlhc)?  I don't mean to have you do all the work for me, but I would appreciate a hint of how these pieces all fit together (or if the ecosystem actually isn't there yet, or if what I'm expecting to be able to do is really not what I2C offers).

I've done the installs, so I can execute this program (though it doesn't fully operate until my parts arrive, so perhaps there's a bug yet to be hit):

```
use libftd2xx::{library_version, Ftdi, FtdiCommon, FtStatus, };

fn main() -> Result<(), FtStatus> {
    println!("Library Version: {}", library_version()?);

    let mut ft = Ftdi::new()?;
    let info = ft.device_info()?;
    let version = ft.driver_version()?;
    println!("Driver Version: {}", version);
    println!("Device information: {:?}", info);

    return Ok(());
}
```

**Comment by newAM** [comment]
I do have this other crate which implements the embedded SPI/GPIO traits, unfortunately I still need to add I2C to it:
https://github.com/newAM/ftd2xx-embedded-hal

I lost a bit of interest because of this issue, https://github.com/newAM/libftd2xx-rs/issues/13, though I could work around that.

If you're on Linux there is also this project that uses the open source libftdi driver, and it has I2C:
https://github.com/geomatsi/ftdi-embedded-hal

**Comment by JasonKleban** [comment]
How could someone help?

**Comment by newAM** [comment]
It is impelmenting the protocol.  This crates provides all the building blocks, but there is implementation work to be done (in the ftd2xx-embeded-hal crate) to build the I2C functions out of the various MPSSE commands.

**Comment by newAM** [comment]
Small update, with #13 resolved I'm picking this up again.  I have more hardware on the way (my current FTDI chips are preoccupied in projects).  I should have a small I2C demo in a few weeks.

**Comment by newAM** [comment]
There's some preliminary code here now (unreleased):

* Implementation: https://github.com/newAM/ftd2xx-embedded-hal/blob/main/src/i2c.rs
* Usage: https://github.com/newAM/ftd2xx-embedded-hal/blob/main/examples/bme280.rs

It works for me, but I want to get my logic analyzer out and take a look at the signals before I make a release of `ftd2xx-embedded-hal`.

That is quite a bit of code to drive the MPSSE for I2C; I will not be adding an example to this repository, otherwise it becomes a chore to keep code in sync, but I will add a link so people looking at this crate can find that I2C code.

I will keep this issue open until:

* [ ] A release of `ftd2xx-embedded-hal` is made with the I2C traits.
* [ ] Links to the `ftd2xx-embedded-hal` crate are added in this crate.

**Comment by newAM** [comment]
Now available in ftd2xx-embedded-hal v0.5.0: https://crates.io/crates/ftd2xx-embedded-hal/0.5.0

**Comment by circuit4u-medium** [comment]
some I2C gist

https://gist.github.com/circuit4u-medium/a9d8a757cdcf3d19f8ca06848af5c22d
